pandas/core/groupby/generic.py

```diff
@@ -1397,7 +1398,11 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
         op = GroupByApply(self, func, args=args, kwargs=kwargs)
         result = op.agg()
         if not is_dict_like(func) and result is not None:
-            return result
+            # GH #52849
+            if is_list_like(func) and not self.as_index:
```

Comment:
Can you switch the order of these two checks; best to have the fastest check first for short-circuiting.